mlir/test/Dialect/Linalg/transform-tile-and-fuse-pack-unpack.mlir

```diff
@@ -0,0 +1,121 @@
+// RUN: mlir-opt %s --transform-interpreter --split-input-file -canonicalize | FileCheck %s
+
+// For pack op, we use lowerPadLikeWithInsertSlice = false to ensure no insert_slice is generated.
+// This allows linalg.transpose to be fused as a producer operation. Alternatively, without this attribute
+// insert_slice will be generated and fusion blocked.
+
+module {
+  // CHECK-label: func @fuse_pack_as_producer
+  // CHECK:       scf.forall {{.*}} {
+  // CHECK:         linalg.transpose
```

Comment:
Btw, this will basically copy `linalg.transpose` inside `scf.forall` - what makes this beneficial?

Reply:
I think @Max191 can potentially give you more details if you are interested. The goal here is not to have the producer fusion flow disrupted. So potentially we can chain additional ops before transpose when enabled. Follow-up PR may come up to further the progress on this direction.